x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/single_page_layout/index.tsx

```diff
@@ -397,12 +395,12 @@ export const CreatePackagePolicySinglePage: CreatePackagePolicyParams = ({
         });
       } else {
         if (isVerificationError(error)) {
+          setFormState('VALID'); // don't show the add agent modal
           const forceInstall = await confirmForceInstall(packagePolicy.package!);
 
           if (forceInstall) {
-            onSubmit({ force: true });
-          } else {
-            setFormState('VALID');
+            // skip creating the agent policy because it will have already been successfully created
+            onSubmit({ overrideCreatedAgentPolicy: createdPolicy, force: true });
```

Comment:
I have to pass the created agent policy here to prevent us creating it again.

We do store the agent policy in state, however because we call the function from itsself, it seems the state doesnt have time to update and re-generate the useCallback that this function is wrapped in, this meant that it was getting a stale view of the state.